### ISSUE TITLE
A high learning rate may cause a nan or an inf loss with tf.keras.optimizers.SGD

### ISSUE THREAD

**Issue by gdhy9064** [tensorflow/tensorflow#38796]
There is a sample which would cause a nan or an inf loss in TFv2.2.0-rc2.
```
import numpy as np
import tensorflow as tf
from tensorflow.keras import Model
from tensorflow.keras.applications import VGG16
from tensorflow.keras.layers import *

vgg = tf.keras.applications.VGG16(include_top=False, input_shape=(300, 400, 3))
o = Conv2D(512, [1, 1])(vgg.output)
l = tf.keras.losses.MSE(vgg.output, o)
m = Model(vgg.input, l)
m.add_loss(l)
m.compile(optimizer=tf.keras.optimizers.SGD(learning_rate=1e-3))
m.fit(np.ones([2, 300, 400, 3]), epochs=10)
```
When I decrease the learning rate, it is of less probability to happen.

Code correction: 
```
m.add_loss(o)
```
to 
```
m.add_loss(l)
```

**Comment by Saduf2019** [comment]
i am able to replicate the issue, please find the [gist here](https://colab.sandbox.google.com/gist/Saduf2019/fc8fde884361678b096f5c9e85bddf78/38776.ipynb). Thanks!

**Comment by jvishnuvardhan** [comment]
@gdhy9064 Can you please tell us little more details about your use case that require this architecture? Thanks!

**Comment by gdhy9064** [comment]
@jvishnuvardhan I use the similar structure trying to build a faster-rcnn. And the problem here doesn't influence me a lot, I can use another optimizer instead. I am just confused that the high learning rate with SGD would have such a strange behavior. Thanks for your response.

**Comment by tanzhenyu** [comment]
@gdhy9064 High learning rate is usually the root cause for many NAN problems. You can try with a lower value, or with another adaptive learning rate optimizer such as Adam.

**Comment by gdhy9064** [comment]
@tanzhenyu Very sorry for the typos in the sample, the loss should be the varible l, not varible o. I have corrected them in the [gist here](https://colab.research.google.com/gist/gdhy9064/6d0026d2eda9bcd5c1f2dd493cb4d4a8/-38776-ipynb.ipynb). I am not sure whether it is just the problem of the learning rate. Thanks.

**Comment by ravikyram** [comment]
@gdhy9064 
I tried in colab with TF nightly version(`2.4.0-dev20200805`) and i am not seeing any issue.Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/b5b4078213e10dd15fc942ed7b52b91f/untitled221.ipynb).
Please, verify once and close the issue if resolved.Thanks!

**Comment by gdhy9064** [comment]
@ravikyram I have tried with TF nightly version (2.4.0-dev20200805), the problem still exists. Maybe the large gradient during training is the cause. Thanks for your kindly reply. I will close the issue now.
